Dockerfile

```diff
@@ -31,11 +31,27 @@ RUN        true \
            && rm -rf /root/.npm \
            && true
 
+# Build a "files" layer
+#
+# This layer is populated with up-to-date files from
+# Calypso development.
+#
+# If package.json and npm-shrinkwrap are unchanged,
+# `install-if-deps-outdated` should require no action.
+# However, time is being spent in the build step on
+# `install-if-deps-outdated`. This is because in the
+# following COPY, the npm-shrinkwrap mtime is being
+# updated, which is confusing `install-if-deps-outdated`.
+# Touch after copy to ensure that this layer will
+# not trigger additional install as part of the build
+# in the following step.
+COPY       . /calypso/
+RUN        touch npm-shrinkwrap.json node_modules
```

Comment:
`touch`ing just `node_modules` would be better. `install-if-deps-updated` doesn't do any work if `node_modules` is **newer** than `shrinkwrap`.

Reply:
👍  Nice, thanks. 28881e4cf89228e4dd259d62730493317a80a475